sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -0,0 +1,369 @@
+    
+  Write-Host "$($indent)-ResourceGroupName`t`tThe name of the Azure Resource Group that will contain the resources"
+  Write-Host "$($indent)`t`t`t`tused for the tests.  This will be created if it does not exist."
+  Write-Host ""
+
+  Write-Host "$($indent)-EventHubsName`t`tThe name of the Event Hubs namespace that will host the "
+  Write-Host "$($indent)`t`t`t`tEvent Hub instances for testing.  This will be created if it does"
+  Write-Host "$($indent)`t`t`t`tnot exist."
+  Write-Host ""
+
+  Write-Host "$($indent)-ServicePrincipalName`tThe name to use for the service principal that will be"
+  Write-Host "$($indent)`t`t`t`tbe created to manage the Event Hub instances dynamically for the tests.  This"
+  Write-Host "$($indent)`t`t`t`tprincipal must not already exist."
+  Write-Host ""
+
+  Write-Host "$($indent)-AzureRegion`t`tThe Azure region that resources should be created in.  This value should be"
+  Write-Host "$($indent)the name of the region, in lowercase, with no spaces.  For example: southcentralus"
+  Write-Host ""
+  Write-Host "$($indent)`t`t`t`tDefault: South Central US (southcentralus)"
+  Write-Host ""  
+}
+
+function SelectRandomCharacters
+{    
+    <#
+      .SYNOPSIS
+        Selects a number of random characters from a set.
+    #>
+
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        [int]$length, 
+
+        [Parameter(Mandatory=$true)]
+        [string]$characters
+    ) 
+
+    $random = (1..$length | ForEach { Get-Random -Maximum $characters.length })
+    return (-join $characters[$random])
+}
+
+function GenerateRandomPassword()
+{
+    <#
+      .SYNOPSIS
+        Generates a random password that can be assigned to a service principal.
+        
+      .DESCRIPTION
+        The password generated by this function will contain a mix of alpha,
+        numeric, and special characters and will vary a small amount in its length.        
+    #>
+
+    $baseLength = (Get-Random -Minimum 28 -Maximum 37)      
+    $upper = (Get-Random -Minimum 6 -Maximum ([int][Math]::Ceiling($baseLength / 3)))  
+    $special = (Get-Random -Minimum 2 -Maximum 3)     
+    $lower = ($baseLength - $upper - $special - $special)                                 
+
+    $password = SelectRandomCharacters $lower "abcdefghiklmnoprstuvwxyz"
+    $password += SelectRandomCharacters $upper "ABCDEFGHKLMNOPRSTUVWXYZ"
+    $password += SelectRandomCharacters $special "1234567890"
+    $password += SelectRandomCharacters $special "!$%&/()=?}][{@#*+"
+
+    $scrambled = ($password.ToCharArray()) | Get-Random -Count ($password.Length)
+    return (-join $scrambled)
+}
+
+function TearDownResources
+{
+    <#
+      .SYNOPSIS
+        Cleans up any Azure resoruces created by the script.
+        
+      .DESCRIPTION
+        Responsible for cleaning up any Azure resoruces created 
+        by the script in case of failure.
+    #>
+    
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        [bool] $cleanResourceGroup,

```

Comment:
nit: I prefer to use `[Switch]` than boolean. You can do:
`TearDownResources -CleanEventHubs`, to say you want to clean up the event hubs but not the resource group since the switch will not be defined.

Reply:
Never mind. this is not a user facing function. I see how it is used.

Reply:
Yeah, given the variance, it wasn't something that I could use switch for easily.